### PR TITLE
[9.0][MIG] product analytic

### PR DESCRIPTION
**Pull request by grindtildeath**
Superseeds (according to author's comment) : https://github.com/OCA/account-analytic/pull/67

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please squash together "OCA Transbot commits": https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

**Comment by grindtildeath** [comment]
@pedrobaeza Done

File: product_analytic/account_invoice.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+from openerp import models, api
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    @api.multi
+    def product_id_change(
+            self, product, uom_id, qty=0, name='', type='out_invoice',
```
Comment (grindtildeath) [inline product_analytic/account_invoice.py]: :warning: type built-in

File: product_analytic/account_invoice.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                account_analytic_id =\
+                    product_o.expense_analytic_account_id.id or\
+                    product_o.categ_id.expense_analytic_account_id.id
+            if account_analytic_id:
+                res['value']['account_analytic_id'] = account_analytic_id
+        return res
+
+    @api.model
+    def create(self, vals):
+        type = self.env.context.get('inv_type', 'out_invoice')
```
Comment (grindtildeath) [inline product_analytic/account_invoice.py]: :warning: type built-in

File: product_analytic/account_invoice.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            price_unit=price_unit,
+            currency_id=currency_id, company_id=company_id)
+        if company_id is None:
+            company_id = self._context.get('company_id', False)
+        self = self.with_context(
+            company_id=company_id, force_company=company_id)
+        if product:
+            product_o = self.env['product.product'].browse(product)
+            account_analytic_id = False
+            if type in ('out_invoice', 'out_refund'):
```
Comment (grindtildeath) [inline product_analytic/account_invoice.py]: :warning: type built-in

**Review by grindtildeath — COMMENTED** [review]
I should fix this, but reviews would still be welcome.

**Comment by grindtildeath** [comment]
@yvaucher @TDu @alexis-via  @pedrobaeza A small review to get this merged would be much appreciated !

**Pull request closed without merging**